src/main/scala/org/ergoplatform/http/api/TransactionsApiRoute.scala

```diff
@@ -68,4 +73,41 @@ case class TransactionsApiRoute(readersHolder: ActorRef, nodeViewActorRef: Actor
     ApiResponse(getUnconfirmedTransactions(offset, limit))
   }
 
+  val feeHistParam: Directive[(Int, Long)] = parameters("bins".as[Int] ? 10, "maxtime".as[Long] ? (60*1000L))
+
+  def getFeeHistogram(nBins : Int, maxWaitTimeMsec: Long, wtxs : Seq[WeightedTxId]): Array[FeeHistogramBin] = {
+    val histogram = Array.fill(nBins + 1)(FeeHistogramBin(0,0))
+    val now = System.currentTimeMillis()
+    val interval = maxWaitTimeMsec / nBins
+    for (wtx <- wtxs) {
+      val waitTime = now - wtx.created
+      val bin = if (waitTime < maxWaitTimeMsec) (waitTime/interval).asInstanceOf[Int] else nBins
+      histogram(bin).nTxns += 1
+      histogram(bin).totalFee += wtx.feePerKb
+    }
+    histogram
+  }
```

Comment:
This method can be (and IMO should be) moved out of this route class (for example into some builder class). The time provider can then be abstracted and the algorithms is unit tested (including providing test vectors for regression, which is necessary for the public API)